Prevent Junit 4 tests from being skipped due to jupiter engine

jtnord opened the pull request:
@jglick suggested before when this happened we should have a check but I
never implemented it.
now it has happened again lets add the check to make sure tests do not
get skipped without anyone realising because we accidently pull in
jupiter-engine.

this should become green when #5712  is merged

See also 
<!-- Comment: 
A great PR typically begins with the line below.
Replace XXXXX with the numeric part of the issue's id you created on JIRA.
Please note that if you want your changes backported into LTS, you will need to create a JIRA ticket for it. Read https://www.jenkins.io/download/lts/#backporting-process for more.
-->
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

<!-- Comment: 
If the issue is not fully described in the ticket, add more information here (justification, pull request links, etc.).

 * We do not require JIRA issues for minor improvements.
 * Bugfixes should have a JIRA issue (backporting process).
 * Major new features should have a JIRA issue reference.
-->

### Proposed changelog entries

N/A

<!-- Comment: 
The changelogs will be integrated by the core maintainers after the merge.  See the changelog examples here: https://www.jenkins.io/changelog/ -->

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

<!-- For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc) -->

### Desired reviewers

@mention

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/code-reviewers
-->

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

jtnord committed:
Prevent Junit 4 tests from being skipped due to jupiter engine

@jglick suggested before when this happened we should have a check but I
never implemented it.
now it has happened again lets add the checj to make sure tests do not
get skipped withouth anyone realising because we accidently pull in
jupiter-engine.

MarkEWaite commented on core/src/test/java/jenkins/Junit4TestsRanTest.java:
```diff
@@ -0,0 +1,13 @@
+    @Test
+    void anything() {
+        // intentionally blank.
+        // we just want a test that runs with junit 4 so that if tests get skipped due to the introduction of ju[iter engine we are allerted
```
```suggestion
        // we just want a test that runs with junit 4 so that if tests get skipped due to the introduction of jupiter engine we are alerted
```

jtnord committed:
Update core/src/test/java/jenkins/Junit4TestsRanTest.java

Co-authored-by: Mark Waite <mark.earl.waite@gmail.com>

jtnord commented:
waiting for a green build now that #5712 is merged, then should be ready-for-merge

jglick reviewed (approved):
Thanks!

jtnord committed:
test case needs to be public!

jtnord commented on core/src/test/java/jenkins/Junit4TestsRanTest.java:
```diff
@@ -0,0 +1,13 @@
+
+import org.junit.Test;
+
+class Junit4TestsRanTest {
```
it helps if the test class is `public`  🤦

jtnord committed:
the missing tests where from the test package

The tests not being executied due to #5703 where in the test package, so
add a test and check there too.
